crates/bevy_ecs/src/world/mod.rs

```diff
@@ -1303,8 +1311,21 @@ impl World {
     #[inline]
     #[track_caller]
     pub fn insert_resource<R: Resource>(&mut self, value: R) {
-        #[cfg(feature = "track_change_detection")]
-        let caller = core::panic::Location::caller();
+        self.insert_resource_with_caller(
+            value,
+            #[cfg(feature = "track_change_detection")]
+            Location::caller(),
+        );
+    }
+
+    /// Split into a new function so we can pass the calling location into the function when using
+    /// as a command.
```

Comment:
```suggestion
    // Split into a new function so we can pass the calling location into the function when using
    // as a command.
```

(Non-blocking) These comments are more notes than documentation, so I think `//` works better in this case.

Reply:
I don't see why you wouldn't want to have this show up in the docs/IDE. It's public only to the crate, so the only people seeing this will be users of bevy_ecs, in which case this is useful context when deciding which function to use.